tools/functional-tester/etcd-runner/command/election_command.go

```diff
@@ -65,18 +71,30 @@ func runElectionFunc(cmd *cobra.Command, args []string) {
 				break
 			}
 		}
-		e := concurrency.NewElection(s, "electors")
 
-		rcs[i].acquire = func() error {
-			<-releasec
+		e := concurrency.NewElection(s, election)
+		rcs[i].acquire = func() (err error) {
 			ctx, cancel := context.WithCancel(context.Background())
+			donec := make(chan struct{})
 			go func() {
-				if ol, ok := <-e.Observe(ctx); ok {
+				defer close(donec)
+				ol, ok := <-e.Observe(ctx)
```

Comment:
If I understand the observe code correctly, Observe will watch on leader's mod rev at here https://github.com/coreos/etcd/blob/master/clientv3/concurrency/election.go#L203. If there is a leader and mod rev < compacted rev, the wch will always be closed regardless how many time I call Observe. Correct me if I am wrong.